bound/bound.go

```diff
@@ -16,23 +16,40 @@ func (b Bound) IsValid() bool {
 	return b.Min <= b.Max
 }
 
-// Intersection returns a Bound that is the intersection of the input bounds.
-// If the intersection is empty or if the input length is zero,
-// then the NaN Bound will be returned.
+// Intersection returns the intersection of the input bounds if possible.
+// Otherwise a NaN Bound is returned.
 func Intersection(bounds ...Bound) Bound {
 	if len(bounds) == 0 {
 		return Bound{Min: math.NaN(), Max: math.NaN()}
 	}
 
-	ret := Bound{Min: bounds[0].Min, Max: bounds[0].Max}
+	intersection := Bound{Min: bounds[0].Min, Max: bounds[0].Max}
 	for _, b := range bounds[1:] {
-		ret.Min = math.Max(ret.Min, b.Min)
-		ret.Max = math.Min(ret.Max, b.Max)
+		intersection.Min = math.Max(intersection.Min, b.Min)
+		intersection.Max = math.Min(intersection.Max, b.Max)
 	}
 
-	if !ret.IsValid() {
+	if !intersection.IsValid() {
 		return Bound{Min: math.NaN(), Max: math.NaN()}
 	}
 
-	return ret
+	return intersection
+}
+
+// Union returns the contiguous union of the input bounds if possible.
+// Otherwise a NaN Bound is returned.
+func Union(bounds ...Bound) Bound {
+	if len(bounds) == 0 {
+		return Bound{Min: math.NaN(), Max: math.NaN()}
+	}
+
+	union := Bound{Min: bounds[0].Min, Max: bounds[0].Max}
+	for _, b := range bounds[1:] {
+		if b.Max < union.Min || union.Max < b.Min {
+			return Bound{Min: math.NaN(), Max: math.NaN()}
+		}
+		union.Min = math.Min(union.Min, b.Min)
+		union.Max = math.Max(union.Max, b.Max)
+	}
```

Comment:
For consistency with the behaviour of `Intersection`, we should probably convert invalid values at this point to NaNs. At the moment, if one of the inputs happens to have one NaN end and one non-NaN end, the output here will be the same unless it's otherwise invalid.